include/mbgl/map/annotation.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef MBGL_MAP_ANNOTATIONS
+#define MBGL_MAP_ANNOTATIONS
+
+#include <mbgl/map/map.hpp>
```

Comment:
This leads to not seeing `AnnotationSegment`, and that can't be forward declared because it is used in `getPoint()`. That method can't be moved inside of `annotation.cpp` because then you get a redefinition error between the forward declaration of `AnnotationSegment` and the one brought into `annotation.cpp` with `#include <mbgl/map/map.hpp>`. This went really deep and doing it like this seemed the only way to keep things happy.